web/src/app/shared/tree/tree.component.ts

```diff
@@ -160,9 +189,25 @@ export class TaskanaTreeComponent implements OnInit, AfterViewChecked, OnDestroy
         event.target.localName === 'taskana-tree')
   }
 
+  private getClassification (classificationId: string): Promise<ClassificationDefinition> {
+    return this.classificationsService.getClassification(classificationId);
+  }
+
+  private async updateClassification(classification: Classification) {
+    await this.classificationsService.putClassification(classification._links.self.href, classification);
+    this.refreshClassification.emit(classification.key);
+  }
+
+  private collapseParentNodeIfItIsTheLastChild(node: any) {
+    if (node.parentId.length > 0 && this.getNode(node.parentId).children.length < 2) {
+      this.getNode(node.parentId).collapse();
```

Comment:
I am not sure if this collapse is synchronous, maybe it is needed to refrest the ui before continue, it could cause strange behaviour